syntax = "proto3";

package calculator;

// Calculator service definition.
service Calculator {
  // Add two numbers.
  rpc Add (CalculationRequest) returns (CalculationResponse);

  // Subtract two numbers.
  rpc Subtract (CalculationRequest) returns (CalculationResponse);

  // Multiply two numbers.
  rpc Multiply (CalculationRequest) returns (CalculationResponse);

  // Divide two numbers.
  rpc Divide (CalculationRequest) returns (CalculationResponse);
}

// Request message containing two numbers.
message CalculationRequest {
  double number1 = 1;
  double number2 = 2;
}

// Response message containing the result.
message CalculationResponse {
  double result = 1;
}